src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureFlagV1KeyValueAdapter.cs

```diff
@@ -11,7 +11,7 @@
 
 namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.FeatureManagement
 {
-    internal class FeatureManagementKeyValueAdapter : IKeyValueAdapter
+    internal class FeatureFlagV1KeyValueAdapter : IKeyValueAdapter
```

Comment:
I expected version to be a parameter in the adapter constructor rather than being bundled into the type. Is there a motivation to have separate types?

Reply:
If we use a single adapter, we need to keep track of the schema version within the adapter. This means that the adapter must decide at runtime whether the feature flag will be flattened with V1 schema or V2 schema. 
I imagine it would look something like this:

```
public Task<IEnumerable<KeyValuePair<string, string>>> ProcessKeyValue(ConfigurationSetting setting, CancellationToken cancellationToken)
{
    if (schema=="v1")
        return ProcessKeyValueInternalV1(....);
    return ProcessKeyValueInternalV2(....);
}
```

This causes an overhead of the `if` check for each feature flag that is loaded. Having two separate  adapters gives us the option to make that decision only once during startup and add only the necessary adapters.

Reply:
> I imagine it would look something like this

My thoughts too. The if check doesn't seem costly, especially compared to what the process methods are doing. Adding new types to a code base does introduce a non-trivial design complexity.

Reply:
The `if` check will be done once for each feature flag during startup, multiplied by the number of times we update anything in `IConfiguration` due to a refresh operation. We dont know how many feature flags will be loaded by an application so it seems like a significant optimization if we can avoid the additional check. Thoughts?

Reply:
An `if` statement does have non-zero cost, but I can't imagine any observable perf impact by saving on avoiding an `if` statement there.  I wouldn't say that there's much of a reason to avoid it.